File: vben-admin-backend/java-backend/src/main/java/com/vben/backend/module/system/controller/DashboardController.java
package com.vben.backend.module.system.controller;

import com.vben.backend.common.result.R;
import com.vben.backend.module.system.service.DashboardService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 仪表盘接口：提供分析页和工作台统计数据。
 * 所有已登录用户均可访问。
 *
 * @author Starry
 */
@RestController
@RequestMapping("/dashboard")
@RequiredArgsConstructor
public class DashboardController {

    private final DashboardService dashboardService;

    /** GET /dashboard/overview：概览统计 */
    @GetMapping("/overview")
    public R<java.util.Map<String, Object>> overview() {
        return R.ok(dashboardService.overview());
    }

    /** GET /dashboard/user-trends：用户增长趋势 */
    @GetMapping("/user-trends")
    public R<java.util.List<java.util.Map<String, Object>>> userTrends() {
        return R.ok(dashboardService.userTrends());
    }

    /** GET /dashboard/role-distribution：角色分布 */
    @GetMapping("/role-distribution")
    public R<java.util.List<java.util.Map<String, Object>>> roleDistribution() {
        return R.ok(dashboardService.roleDistribution());
    }

    /** GET /dashboard/dept-distribution：部门用户分布 */
    @GetMapping("/dept-distribution")
    public R<java.util.List<java.util.Map<String, Object>>> deptDistribution() {
        return R.ok(dashboardService.deptDistribution());
    }

    /** GET /dashboard/workspace：工作台数据 */
    @GetMapping("/workspace")
    public R<java.util.Map<String, Object>> workspace() {
        return R.ok(dashboardService.workspace());
    }
}

File: vben-admin-backend/node-backend/src/modules/dashboard/dashboard.controller.ts
import { Controller, Get, UseGuards } from '@nestjs/common';
import { ApiTags, ApiOperation, ApiBearerAuth } from '@nestjs/swagger';
import { DashboardService } from './dashboard.service.js';
import { R } from '../../common/result.js';
import { JwtAuthGuard } from '../../common/guards/jwt-auth.guard.js';

@ApiTags('仪表盘')
@ApiBearerAuth('JWT')
@Controller('dashboard')
@UseGuards(JwtAuthGuard)
export class DashboardController {
  constructor(private readonly dashboardService: DashboardService) {}

  @Get('overview')
  @ApiOperation({ summary: '概览统计' })
  async overview() {
    return R.ok(await this.dashboardService.overview());
  }

  @Get('user-trends')
  @ApiOperation({ summary: '用户增长趋势' })
  async userTrends() {
    return R.ok(await this.dashboardService.userTrends());
  }

  @Get('role-distribution')
  @ApiOperation({ summary: '角色分布' })
  async roleDistribution() {
    return R.ok(await this.dashboardService.roleDistribution());
  }

  @Get('dept-distribution')
  @ApiOperation({ summary: '部门用户分布' })
  async deptDistribution() {
    return R.ok(await this.dashboardService.deptDistribution());
  }

  @Get('browser-distribution')
  @ApiOperation({ summary: '浏览器分布' })
  async browserDistribution() {
    return R.ok(await this.dashboardService.browserDistribution());
  }
}
